tools/mpy-tool.py

```diff
@@ -152,21 +152,58 @@ def decode_uint(bytecode, ip):
             break
     return ip, unum
 
+def read_compressed_prelude(read_byte):
+    z = read_byte()
+    # xSSSSEAA
+    n_state = (z >> 3) & 0xf;                                   \
+    n_exc_stack = (z >> 2) & 0x1;                               \
+    scope_flags = 0;                                            \
+    n_pos_args = z & 0x3;                                       \
+    n_kwonly_args = 0;                                          \
+    n_def_pos_args = 0;                                         \
+    n = 0
+    while z & 0x80:
+        z = read_byte()
+        # xFSSKAED
+        n_state |= (z & 0x30) << (2 * n);                       \
+        n_exc_stack |= (z & 0x02) << n;                         \
+        scope_flags |= ((z & 0x40) >> 6) << n;                  \
+        n_pos_args |= (z & 0x4) << n;                           \
+        n_kwonly_args |= ((z & 0x08) >> 3) << n;                \
+        n_def_pos_args |= (z & 0x1) << n;                       \
+        n += 1
+    n_state += 0*n_pos_args + 1;                                  \
```

Comment:
Isn't this just `n_state += 1`?

Reply:
Yes indeed.  I originally had it as `n_state += n_pos_args + 1` to improve compression but it was more complex to decode when only `n_state` is needed.  So this should be simplified.